.gitlab-ci.yml

```
@@ -67,63 +65,108 @@ variables:                         &default-vars
-
-#### stage:                       test
-
-check-code:
-  stage:                           test
-  <<:                              *docker-env
-  <<:                              *common-refs
-  script:
-    - cargo install cargo-hack
-    - cargo hack check --workspace --each-feature
-
-test:
-  stage:                           test
-  <<:                              *docker-env
-  <<:                              *common-refs
-  script:
-    - cargo test
-
-benchmarks:
+# fmt:
+#   stage:                           lint
+#   <<:                              *docker-env
+#   <<:                              *common-refs
+#   script:
+#     - cargo fmt --all -- --check
+
+# clippy:
+#   stage:                           lint
+#   <<:                              *docker-env
+#   <<:                              *common-refs
+#   script:
+#     - cargo clippy --all-targets
+
+# check-rustdoc-links:
+#   stage:                           lint
+#   <<:                              *docker-env
+#   <<:                              *common-refs
+#   script:
+#     - RUSTDOCFLAGS="--deny broken_intra_doc_links" cargo doc --workspace --no-deps --document-private-items
+
+# #### stage:                       test
+
+# check-code:
+#   stage:                           test
+#   <<:                              *docker-env
+#   <<:                              *common-refs
+#   script:
+#     - cargo install cargo-hack
+#     - cargo hack check --workspace --each-feature
+
+# test:
+#   stage:                           test
+#   <<:                              *docker-env
+#   <<:                              *common-refs
+#   script:
+#     - cargo test
+
+.benchmarks: &benchmarks
   stage:                           benchmark
   <<:                              *docker-env
   <<:                              *collect-artifacts
-  <<:                              *schedule-refs
+  <<:                              *common-refs
   timeout:                         3h
   variables:
     MEASUREMENT_TIME:              10
     SLOW_MEASUREMENT_TIME:         60
+    GIT_STRATEGY:                  none
   script:
-    - cargo bench -p jsonrpsee-benchmarks -- --output-format bencher | tee output.txt
-    - mkdir artifacts
-    - cp output.txt artifacts/
+    - export FILENAME="output-${TAG}${CI_RUNNER_DESCRIPTION}.txt"
+    - git clone https://github.com/paritytech/jsonrpsee.git
+    - cd jsonrpsee
+    - git fetch --all --tags
+    - git checkout ${TAG}
+    - cargo bench -p jsonrpsee-benchmarks -- --output-format bencher | tee ${FILENAME}
+    - mkdir -p artifacts
+    - cp ${FILENAME} artifacts/
     - echo ${CI_RUNNER_DESCRIPTION}
     - echo "RUNNER_NAME=${CI_RUNNER_DESCRIPTION}" > runner.env
+
+benchmarks-v14-ci5:
+  <<: *benchmarks
+  variables:
+    MEASUREMENT_TIME:              10
+    SLOW_MEASUREMENT_TIME:         60
+    GIT_STRATEGY:                  none
+    TAG: "v.0.14.0"
```

I haven't backported the changed benches to the tags FWIW

```suggestion
    BRANCH: "v0.14.0-bench"
```